### PR TITLE
add status request to grpc endpoint

### PR DESCRIPTION
**Pull request by stsatlantis**
[sc-14992]

- Move endpoint that registers executions into the `SecurityReportIngestionService`
- Add a new endpoint to register manual triggered executions.

### CODE REVIEW

File: com/coralogix/xdr/ingestion/v1/security_report_ingestion_service.proto [hunk 1]
```diff
@@ -6,19 +6,50 @@ import "com/coralogix/xdr/ingestion/v1/audit_log.proto";
...
-    option (audit_log_description).description = "Endpoints to consume SecurityReports";
+    option (audit_log_description).description = "Endpoint to consume SecurityReports";
     option (google.api.http) = {
       post: "/xdr/v1/security_report"
       body: "security_report"
     };
   }
+
+  rpc GetExecutionStatus(GetExecutionStatusRequest) returns (GetExecutionStatusResponse) {
+    option (audit_log_description).description = "Endpoint to check if the scan needs to run";
```
Comment (xBATx) [inline com/coralogix/xdr/ingestion/v1/security_report_ingestion_service.proto:21]: I'd rephrase it to `Endpoint to check the actual execution status.`

File: com/coralogix/xdr/ingestion/v1/security_report_ingestion_service.proto [hunk 2]
```diff
@@ -6,19 +6,50 @@ import "com/coralogix/xdr/ingestion/v1/audit_log.proto";
...
+  }
+
+  rpc GetManualExecutionStatus(GetManualExecutionStatusRequest) returns (GetManualExecutionStatusResponse) {
+    option (audit_log_description).description = "Endpoint to check if manual scan can run";
```
Comment (xBATx) [inline com/coralogix/xdr/ingestion/v1/security_report_ingestion_service.proto:25]: Why do we need to differentiate between manual and ordinary execution? Couldn't we rely on `GetExecutionStatus` and trigger it based on the `status` - i.e. when it's idle or ready?

**Review by xBATx — COMMENTED** [review]
See some comments and also please check the build as it's failing.

File: com/coralogix/xdr/ingestion/v1/security_report_ingestion_service.proto [hunk 3]
```diff
@@ -6,19 +6,50 @@ import "com/coralogix/xdr/ingestion/v1/audit_log.proto";
...
+  }
+
```
Comment (stsatlantis) [inline com/coralogix/xdr/ingestion/v1/security_report_ingestion_service.proto:27]: We don't need it. as I mentioned the scheduled and the manual is the same.

File: com/coralogix/xdr/ingestion/v1/security_report_ingestion_service.proto [hunk 4]
```diff
@@ -6,19 +6,50 @@ import "com/coralogix/xdr/ingestion/v1/audit_log.proto";
...
   SecurityReport security_report = 1;
 }
 
 message PostSecurityReportResponse {}
+
+message GetManualExecutionStatusRequest {}
+
+message GetManualExecutionStatusResponse {
+  google.protobuf.BoolValue can_run = 1;
+}
```
Comment (stsatlantis) [inline com/coralogix/xdr/ingestion/v1/security_report_ingestion_service.proto:40]: No need for this

File: com/coralogix/xdr/ingestion/v1/security_report_ingestion_service.proto (hunk 2 above)
Comment (StrDragos) [reply com/coralogix/xdr/ingestion/v1/security_report_ingestion_service.proto]: The manual execution is just a marker that the user wants to trigger execution on the next polling event. It needs to be stored in a different table. Thus, the need to differentiate between the 2 calls.

Comment (StrDragos) [reply com/coralogix/xdr/ingestion/v1/security_report_ingestion_service.proto]: BTW this was moved to the `xdr-api` and it's named scanNow.
